src/tests/Microsoft.PowerFx.Interpreter.Tests/Scenarios/DataTableMarshallerProvider.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Data;
+using Microsoft.PowerFx.Interpreter.Tests;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Tests
+{
+    /// <summary>
+    /// Wire up <see cref="DataTable"/> conversions to <see cref="TypeMarshallerCache.Marshal{T}(T)"/>.     
+    /// </summary>
+    internal class DataTableMarshallerProvider : IDynamicTypeMarshaller
+    {
+        public bool TryMarshal(TypeMarshallerCache cache, object value, out FormulaValue result)
+        {
+            if (value is DataTable dataTable)
+            {
+                result = new DataTableValue(dataTable);
+                return true;
+            }
+
+            result = null;
+            return false;
+        }
+    }
+
+    // Wrap a System.Data.DataTable as a Power Fx TableValue
+    // Marshal DataTable as a Table of Records. 
+    // Use column has type info if available; else marshal as untyped object.
```

Comment:
I do not understand this comment

Reply:
System.Data.DataTable has columns. Those columns may be typed (column.DataType is something like Int32 or String) or untyped (column.DataType  is just System.Object)

If it's typed, this marshaller wants to leverage that typeinfo. Ie, Int becomes FormulaType.Number. 
If untyped,  we'll marshal as FormulaType.UntypedObject.

Reply:
"Use column has type info"